server/api/controllers/auth.py

```
@@ -255,3 +276,153 @@ def login():
+
+    # Exchange code for access token
+    token_url = "https://github.com/login/oauth/access_token"
+    token_data = {
+        "client_id": GITHUB_CLIENT_ID,
+        "client_secret": GITHUB_CLIENT_SECRET,
+        "code": code,
+        "redirect_uri": GITHUB_REDIRECT_URI,
+    }
+    token_headers = {"Accept": "application/json"}
+
+    token_res = requests.post(token_url, data=token_data, headers=token_headers).json()
```

Ruff (0.14.8)

Probable use of `requests` call without timeout

(S113)